### ISSUE TITLE
Loading stalled: stop fragment loading results in multiple full player restarts and, finally, freeze

### ISSUE THREAD

**Issue by dananichev** [mangui/flashls#526]
`Loading stalled: stop fragment loading` results in complete player freeze after multiple player restarts.
No 'Error' events can be catched thru ExternalInterface (flashvars => callback). Due to no errors it's not possible to detect this problem via JS and restart the player one more time (or even more than one?).

Log goes here:
[flashls-log.pdf](https://github.com/mangui/flashls/files/261422/flashls-log.pdf)

Any workarounds or suggestions until you will be able to release new version?

**Comment by mangui** [comment]
looks like you are trying to load a 1080p live video with a really bad connection ...
flashls already fires an event on such case : HLSEvent.LIVE_LOADING_STALLED
up to you to listen to it and do whatever you want in such case

**Comment by dananichev** [comment]
@mangui yeah but we have 10 seconds length chunks and each chunk takes about 2-3 seconds to load (with maximum of 5 seconds). Isn't enough? Also, is HLSEvent.LIVE_LOADING_STALLED available in JS's callback function?

**Comment by mangui** [comment]
no it is not reported, feel free to add it and eventually PR

**Comment by dananichev** [comment]
@mangui tbh i removed Flex and right now i have no time for setting whole build environment, but i understand. And what about my first question?

**Comment by dananichev** [comment]
@mangui considering other issues related to `loading stalled` event (i mean in this bugtracker) -- maybe your algorithm is not working correctly? As i wrote before, with 10 seconds chunks 2-5 seconds loading time is more than enough for consistent playback. 

Also i'm not AS developer and i'm not exactly sure where to look for passing events to JS's ExternalInterface (there are `HLSEvent` class, but it has no `ExternalInterface` communication, there are `ChromelessPlayer` class, but there are no list of events to pass to `ExternalInterface`). Any help? 

...

Nvm i'm partially blind. Found it. Anyways is there anything we can do about player blinking after this event? I mean, why it's not possible to just discard one chunk and download next one -> play next one w/o destroying the player completely?

**Comment by mangui** [comment]
in the log you sent me, it takes so long to load 1463060313, that once its loading is finished, 1463060313 slided out of the playlist, now starting @ 1463060322

Loading **1463060313** of **[1463060312,1463060316]**,level 0
Loaded 1463060313 of **[1463060322,1463060326]**,level 0 m/M PTS:8792743/8802723
ERROR:updateFragment:seqnum 1463060313 not found!
WARN:loading stalled:stop fragment loading

in such scenario, there is nothing flashls could do I am affraid.

**Comment by dananichev** [comment]
As a workaround: https://github.com/mangui/flashls/pull/528

But i must ask: why flashls stops downloading chunks after some time at all? I mean, it happens after 2+ live loading stalled errors.

**Comment by pratikpradhan** [comment]
mangui,
How to restart playback?
Is it just loading master.m3u8 again?

**Comment by ajay-jose** [comment]
Hi Pratik 

We are looking for developers with player and media experience. Would you be interested to join? 

Ajay

**Comment by neilrackett** [comment]
Internally, flashls generally uses `_hls.stream.seek(-1);` to restart playback, so that's probably a good place to start.
